dataeng/jobs/analytics/SnowflakeReplicaImportFromS3.groovy

```diff
@@ -1,37 +1,91 @@
-                    stringParam('DATABASE', db_config.get('DATABASE'), 'Name of MySQL database to copy from.')
-                    stringParam('INCLUDE', db_config.get('INCLUDE'), 'List of regular expressions matching table names that should be copied.')
-                    stringParam('EXCLUDE', db_config.get('EXCLUDE'), 'List of regular expressions matching table names that should not be copied.')
+                    stringParam('RUN_DATE', 'today', 'Run date for the job. A string that can be parsed by the GNU coreutils "date" utility.')
+                    stringParam('OVERWRITE', '--overwrite', 'Set to: --overwrite if you want to enable overwrite.')
+                    stringParam('SNOWFLAKE_CREDENTIALS', allVars.get('SNOWFLAKE_CREDENTIALS'), 'The path to the Snowflake credentials file.')
+                    stringParam('WAREHOUSE', 'LOADING_SQOOP', 'The Snowflake warehouse to use.')
+                    stringParam('ROLE', 'PIPELINE_ETL_LOADER_ROLE', 'The Snowflake role to assume.')
+                    stringParam('SNOWFLAKE_DATABASE', 'PROD', 'The destination database in Snowflake.')
+                    stringParam('SCHEMA', jobConfig.get('SCHEMA'), 'The destination raw schema in Snowflake.')
+                    stringParam('SCRATCH_SCHEMA', jobConfig.get('SCRATCH_SCHEMA'), 'Scratch schema name - temporary loading location in Snowflake.')
+                    stringParam('DATABASE', jobConfig.get('DATABASE'), 'Name of MySQL database to copy from.')
+                    stringParam('INCLUDE', jobConfig.get('INCLUDE', ''), 'List of regular expressions matching table names that should be copied.')
+                    stringParam('EXCLUDE', jobConfig.get('EXCLUDE', ''), 'List of regular expressions matching table names that should not be copied.')
+                    stringParam('SQOOP_START_TIME', '', 'The start time of the upstream sqoop load (in ISO timestamp format).')
+                }
+                environmentVariables {
+                    env('APP_NAME', appName)
                 }
                 multiscm common_multiscm(allVars)
                 wrappers common_wrappers(allVars)
-                publishers common_publishers(allVars)
                 steps {
                     shell(dslFactory.readFileFromWorkspace('dataeng/resources/snowflake-replica-import.sh'))
                 }
+                postBuildSteps {
+                    // If we skipped today's run, set the current build to ABORTED so that we don't trigger stitch
+                    // validation.
+                    conditionalSteps {
+                        condition { fileExists('${WORKSPACE}/build_skipped') }
+                        steps { setBuildResult('ABORTED') }
+                    }
+                }
+                publishers common_publishers(allVars)
+                publishers {
+                    downstreamParameterized {
+                        trigger("validate-stitch-against-sqoop" )
```

Comment:
This needs to change once https://openedx.atlassian.net/browse/DENG-360 is complete.